dist/images/Dockerfile.fedora

```diff
@@ -9,25 +9,93 @@
 # are built locally and included in the image (instead of the rpm)
 #
 
-FROM fedora:41
+ARG OVN_FROM=koji
+
+FROM fedora:41 AS ovnbuilder
 
 USER root
 
 ENV PYTHONDONTWRITEBYTECODE yes
 
+# Install tools that are required for building ovs/ovn.
+RUN INSTALL_PKGS="git rpm-build dnf-plugins-core" && \
+    dnf install --best --refresh -y --setopt=tsflags=nodocs $INSTALL_PKGS
+
+# Clone OVS Source Code.
+ARG OVS_REPO=https://github.com/openvswitch/ovs.git
+ARG OVS_BRANCH=main
+WORKDIR /root
+RUN git clone $OVS_REPO --single-branch --branch=$OVS_BRANCH
+
+# Build OVS rpms.
+WORKDIR /root/ovs
+RUN sed -e 's/@VERSION@/0.0.1/' rhel/openvswitch-fedora.spec.in > /tmp/ovs.spec
+RUN dnf builddep -y /tmp/ovs.spec
+RUN ./boot.sh
+RUN ./configure
+RUN make rpm-fedora
+RUN rm rpm/rpmbuild/RPMS/x86_64/*debug*
+RUN rm rpm/rpmbuild/RPMS/x86_64/*devel*
+RUN git log -n 1
+
+# Clone OVN Source Code.
+ARG OVN_REPO=https://github.com/ovn-org/ovn.git
+ARG OVN_BRANCH=main
+ARG OVN_COMMIT=HEAD
+WORKDIR /root
+RUN git clone $OVN_REPO --single-branch --branch=$OVN_BRANCH && \ 
```

Comment:
I read up that 
```
git init
git remote add origin ...
git fetch origin <ref>
git reset FETCH_HEAD
```
might be more flexible approach to getting just what you want

Reply:
Ahh that's waaay better, I was looking for something like that, this way I can pass commits, tags or branches.

Reply:
The difference is that you fetch the whole `origin` instead of the single `OVN_BRANCH`.  But if we're ok with that, that works.

Reply:
But if that's a concern, we should be using `--depth 1` anyways

Reply:
Also true.  For me it works either way.